pkg/csi/service/wcpguest/controller.go

```diff
@@ -460,8 +463,10 @@ func controllerPublishForBlockVolume(ctx context.Context, req *csi.ControllerPub
 					ClaimName: req.VolumeId,
 				},
 			}
+			virtualMachineLock.Lock()
```

Comment:
can we use defer instead of unlocking later?

Reply:
Using defer might significantly reduce the performance. Let's unlock soon after update call